src/io/ctrees_utils.c

```diff
@@ -422,24 +435,18 @@ int fix_upid(const int64_t totnhalos, struct halo_data *forest, struct additiona
                     break;
                 }
             }
-            XASSERT( found == 1, EXIT_FAILURE,
+            XRETURN( found == 1, -1,
                      "Error: Could not locate FOF halo for halo with id = %"PRId64" and upid = %"PRId64"\n",
                      info[i].id, upid);
         }
         if(verbose) {
             fprintf(stderr,"found FOF halo for halnum = %"PRId64". loc = %"PRId64" id = %"PRId64" upid = %"PRId64"\n",
                     i, loc, info[loc].id, info[loc].upid);
         }
-        XASSERT(loc >=0 && loc < totnhalos, EXIT_FAILURE,
+        XRETURN(loc >=0 && loc < totnhalos, -1,
```

Comment:
(This is me deleting a comment because I accidentally told Firefox to not allow additional dialogs which stops me from deleting comments...)